system-x/services/splunk/src/main/java/software/tnb/splunk/service/openshift/Splunk.java

```diff
@@ -0,0 +1,220 @@
+            throw new RuntimeException("Unable to read splunk-operator-namespace.yml or splunk-crds.yaml resource: ", e);
+        }
+
+        Map<String, Object> splunkInstance = Map.of(
+            "apiVersion", "enterprise.splunk.com/v3",
+            "kind", "Standalone",
+            "metadata", Map.of(
+                "name", "s1")
+        );
+        try {
+            OpenshiftClient.get().customResource(createSplunkContext()).inNamespace(OpenshiftConfiguration.openshiftNamespace()).delete();
+            OpenshiftClient.get().customResource(createSplunkContext()).inNamespace(OpenshiftConfiguration.openshiftNamespace())
+                .create(splunkInstance);
+        } catch (IOException e) {
+            throw new RuntimeException("Unable to create integration platform: ", e);
+        }
+    }
+
+    @Override
+    public void undeploy() {
+
+        LOG.info("Undeploying integration resources");
+        OpenshiftClient.get().customResource(createSplunkContext()).inNamespace(OpenshiftConfiguration.openshiftNamespace()).delete();
+        WaitUtils.waitFor(() -> !this.isAppDeploy(), "Waiting until Splunk CR is uninstalled.");
+        OpenshiftClient.get().resourceList(createdResources.stream()
+            .filter(res -> !(res instanceof CustomResourceDefinition
+                || res instanceof io.fabric8.kubernetes.api.model.apiextensions.v1beta1.CustomResourceDefinitionVersion
+                || res instanceof ClusterRole))
+            .collect(Collectors.toList())).withPropagationPolicy(DeletionPropagation.BACKGROUND).delete();
+        OpenshiftClient.get().clusterRoleBindings().withName("system:openshift:scc:nonroot").delete();
+
+        //if CR creates PVC's, they need to be deleted. (usage of the finalizer in CR can in some situations (e.g. failure during operator
+        // creation) cause the deletion of CR get stuck)
+        OpenshiftClient.get().persistentVolumeClaims().withLabel("app.kubernetes.io/name", "standalone").delete();
+    }
+
+    @Override
+    public void openResources() {
+        String splunkCert = OpenshiftClient.get().podShell(OpenshiftClient.get().getAnyPod("app.kubernetes.io/instance", "splunk-s1-standalone"))
+            .execute("cat", "/opt/splunk/etc/auth/cacert.pem").getOutput();
+
+        // @formatter:off
+        apiRoute = OpenshiftClient.get().routes().createOrReplace(new RouteBuilder()
+            .editOrNewMetadata()
+                .withName(ROUTE_NAME)
+            .endMetadata()
+            .editOrNewSpec()
+                .withNewTo().withKind("Service").withName("splunk-s1-standalone-service").withWeight(100)
+                .endTo()
+                .withNewPort().withTargetPort(new IntOrString("https-splunkd")).endPort()
+                .withNewTls().withTermination("reencrypt").withDestinationCACertificate(splunkCert).endTls()
+            .endSpec()
+            .build());
+        // @formatter:on
+        WaitUtils.sleep(5000);
```

Comment:
what for?

Reply:
Sometimes, the app besides the URL was not ready right after (there was an application not found in the route). I don't know why because the service log showed that the application was ready.

Reply:
in that case you can use the http client instead of fixed wait time